02/03/2024 21:07:33 - WARNING - __main__ - Process rank: 0, device: cuda:0, n_gpu: 1distributed training: True, 16-bits training: False
loading checkpoint: /data/ai/ChatGLM2-6B/ptuning/output/adgen-chatglm2-6b-pt-128-1e-4/checkpoint-700
/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/load.py:2483: FutureWarning: 'use_auth_token' was deprecated in favor of 'token' in version 2.14.0 and will be removed in 3.0.0.
You can remove this warning by passing 'token=<use_auth_token>' instead.
  warnings.warn(
[INFO|configuration_utils.py:667] 2024-02-03 21:07:34,492 >> loading configuration file /data/ai/model/chatglm2-6b/config.json
[INFO|configuration_utils.py:667] 2024-02-03 21:07:34,494 >> loading configuration file /data/ai/model/chatglm2-6b/config.json
[INFO|configuration_utils.py:725] 2024-02-03 21:07:34,495 >> Model config ChatGLMConfig {
  "_name_or_path": "/data/ai/model/chatglm2-6b",
  "add_bias_linear": false,
  "add_qkv_bias": true,
  "apply_query_key_layer_scaling": true,
  "apply_residual_connection_post_layernorm": false,
  "architectures": [
    "ChatGLMModel"
  ],
  "attention_dropout": 0.0,
  "attention_softmax_in_fp32": true,
  "auto_map": {
    "AutoConfig": "configuration_chatglm.ChatGLMConfig",
    "AutoModel": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForCausalLM": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForSeq2SeqLM": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForSequenceClassification": "modeling_chatglm.ChatGLMForSequenceClassification"
  },
  "bias_dropout_fusion": true,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "ffn_hidden_size": 13696,
  "fp32_residual_connection": false,
  "hidden_dropout": 0.0,
  "hidden_size": 4096,
  "kv_channels": 128,
  "layernorm_epsilon": 1e-05,
  "model_type": "chatglm",
  "multi_query_attention": true,
  "multi_query_group_num": 2,
  "num_attention_heads": 32,
  "num_layers": 28,
  "original_rope": true,
  "pad_token_id": 0,
  "padded_vocab_size": 65024,
  "post_layer_norm": true,
  "pre_seq_len": null,
  "prefix_projection": false,
  "quantization_bit": 0,
  "rmsnorm": true,
  "seq_length": 32768,
  "tie_word_embeddings": false,
  "torch_dtype": "float16",
  "transformers_version": "4.30.2",
  "use_cache": true,
  "vocab_size": 65024
}
[INFO|tokenization_utils_base.py:1821] 2024-02-03 21:07:34,496 >> loading file tokenizer.model
[INFO|tokenization_utils_base.py:1821] 2024-02-03 21:07:34,496 >> loading file added_tokens.json
[INFO|tokenization_utils_base.py:1821] 2024-02-03 21:07:34,496 >> loading file special_tokens_map.json
[INFO|tokenization_utils_base.py:1821] 2024-02-03 21:07:34,496 >> loading file tokenizer_config.json
[INFO|modeling_utils.py:2575] 2024-02-03 21:07:34,539 >> loading weights file /data/ai/model/chatglm2-6b/pytorch_model.bin.index.json
[INFO|configuration_utils.py:577] 2024-02-03 21:07:34,540 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}

Loading checkpoint shards:  57%|█████████████████████████████████████████████████████▋                                        | 4/7 [00:02<00:02,  1.37it/s]
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:04<00:00,  1.44it/s]
[INFO|modeling_utils.py:3295] 2024-02-03 21:07:39,411 >> All model checkpoint weights were used when initializing ChatGLMForConditionalGeneration.
[WARNING|modeling_utils.py:3297] 2024-02-03 21:07:39,411 >> Some weights of ChatGLMForConditionalGeneration were not initialized from the model checkpoint at /data/ai/model/chatglm2-6b and are newly initialized: ['transformer.prefix_encoder.embedding.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
[INFO|modeling_utils.py:2927] 2024-02-03 21:07:39,412 >> Generation config file not found, using a generation config created from the model config.
prefix: 你是蚂蚁金服公司的客服人员,用清晰简洁的语言回答老年人的问题.如果因为知识库受限无法回答,只需回答对不起，根据参考资料无法回答
eval dataset: Dataset({
    features: ['query', 'answer'],
    num_rows: 315
})
inputs你是蚂蚁金服公司的客服人员,用清晰简洁的语言回答老年人的问题.如果因为知识库受限无法回答,只需回答对不起，根据参考资料无法回答什么是蚂蚁庄园
input_ids [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64790, 64792, 30910, 34607, 41069, 54702, 54764, 33031, 37605, 31698, 30932, 54571, 33724, 40697, 40042, 33287, 52775, 31639, 30930, 31654, 31659, 31848, 55651, 50072, 32067, 33287, 30932, 37757, 33287, 43279, 31123, 31793, 33214, 32315, 32067, 33287, 35318, 41069, 46980]
inputs 你是蚂蚁金服公司的客服人员,用清晰简洁的语言回答老年人的问题.如果因为知识库受限无法回答,只需回答对不起，根据参考资料无法回答什么是蚂蚁庄园
label_ids [64790, 64792, 30910, 41069, 46980, 41680, 33053, 39063, 32488, 39330, 38346, 35439, 32033, 31155]
labels 蚂蚁庄园是一款基于支付宝支付行为的农场养成游戏。
02/03/2024 21:07:42 - INFO - __main__ - *** Evaluate ***
Running tokenizer on validation dataset: 100%|█████████████████████████████████████████████████████████████████████| 20/20 [00:00<00:00, 2729.87 examples/s]
[INFO|trainer.py:577] 2024-02-03 21:07:42,876 >> max_steps is given, it will override any value given in num_train_epochs
[INFO|trainer.py:3200] 2024-02-03 21:07:42,879 >> ***** Running Evaluation *****
[INFO|trainer.py:3202] 2024-02-03 21:07:42,879 >>   Num examples = 20
[INFO|trainer.py:3205] 2024-02-03 21:07:42,879 >>   Batch size = 4
[INFO|configuration_utils.py:577] 2024-02-03 21:07:42,881 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
  0%|                                                                                                                                 | 0/5 [00:00<?, ?it/s][INFO|configuration_utils.py:577] 2024-02-03 21:08:04,105 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
Traceback (most recent call last):
  File "/data/ai/ChatGLM2-6B/ptuning/main.py", line 484, in <module>
    main()
  File "/data/ai/ChatGLM2-6B/ptuning/main.py", line 416, in main
    metrics = trainer.evaluate(metric_key_prefix="eval", do_sample=True, top_p=0.7, max_length=max_seq_length, temperature=0.95)
  File "/data/ai/ChatGLM2-6B/ptuning/trainer_seq2seq.py", line 79, in evaluate
    return super().evaluate(eval_dataset, ignore_keys=ignore_keys, metric_key_prefix=metric_key_prefix)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/trainer.py", line 3053, in evaluate
    output = eval_loop(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/trainer.py", line 3245, in evaluation_loop
    loss, logits, labels = self.prediction_step(model, inputs, prediction_loss_only, ignore_keys=ignore_keys)
  File "/data/ai/ChatGLM2-6B/ptuning/trainer_seq2seq.py", line 203, in prediction_step
    generated_tokens = self.model.generate(**gen_kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/generation/utils.py", line 1572, in generate
    return self.sample(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/generation/utils.py", line 2655, in sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
KeyboardInterrupt
Traceback (most recent call last):
  File "/data/ai/ChatGLM2-6B/ptuning/main.py", line 484, in <module>
    main()
  File "/data/ai/ChatGLM2-6B/ptuning/main.py", line 416, in main
    metrics = trainer.evaluate(metric_key_prefix="eval", do_sample=True, top_p=0.7, max_length=max_seq_length, temperature=0.95)
  File "/data/ai/ChatGLM2-6B/ptuning/trainer_seq2seq.py", line 79, in evaluate
    return super().evaluate(eval_dataset, ignore_keys=ignore_keys, metric_key_prefix=metric_key_prefix)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/trainer.py", line 3053, in evaluate
    output = eval_loop(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/trainer.py", line 3245, in evaluation_loop
    loss, logits, labels = self.prediction_step(model, inputs, prediction_loss_only, ignore_keys=ignore_keys)
  File "/data/ai/ChatGLM2-6B/ptuning/trainer_seq2seq.py", line 203, in prediction_step
    generated_tokens = self.model.generate(**gen_kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/generation/utils.py", line 1572, in generate
    return self.sample(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/generation/utils.py", line 2655, in sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
KeyboardInterrupt